source/cl/tools/oclc/oclc.cpp

```diff
@@ -1391,6 +1392,19 @@ T *oclc::Driver::CastToTypeFloat(const std::vector<std::string> &source,
   return casted_buffers.back().data();
 }
 
+cl_half *oclc::Driver::CastToTypeHalf(const std::vector<std::string> &source,
+                                      vector2d<cl_half> &casted_buffers,
+                                      size_t &size) {
+  std::vector<cl_half> data(source.size());
+  for (size_t i = 0; i < source.size(); ++i) {
+    auto f = atof(source[i].c_str());
+    data[i] = cl_half_from_float(f, cl_half_rounding_mode::CL_HALF_RTE);
```

Comment:
You can make `CastToTypeFloat<cl_half>` work with
```
auto f = atof(source[i].c_str());
if constexpr (std::is_same_v<T, cl_half>) {
  data[i] = cl_half_from_float(f, cl_half_rounding_mode::CL_HALF_RTE);
} else {
  data[i] = static_cast<T>(f);
}
```
so you don't need to duplicate the rest of the function.

Reply:
wonderful, thank you